MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -284,16 +286,36 @@ void vtkSlicerMarkupsToModelLogic::UpdateSelectionNode( vtkMRMLMarkupsToModelNod
 //------------------------------------------------------------------------------
 void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel( vtkMRMLMarkupsToModelNode* markupsToModelModuleNode, vtkPolyData* output )
 {
+  if ( markupsToModelModuleNode == NULL )
+  {
+    vtkErrorMacro( "No markupsToModelModuleNode provided to UpdateOutputCloseSurfaceModel! No operation performed!" );
+    return;
+  }
+
   vtkMRMLMarkupsFiducialNode* markups = markupsToModelModuleNode->GetMarkupsNode();
-  if ( markups == NULL )
+  if ( output == NULL )
   {
+    vtkErrorMacro( "No output poly data provided to UpdateOutputCloseSurfaceModel! No operation performed!" );
     return;
   }
+
+  output->Reset(); // empty the poly data of all existing data
```

Comment:
Memory allocation/deallocation is expensive in general. If it doesn't make the code more complicated, it's better to do it only when needed (e.g., in the branch where you return if there are input markups).